Bugsnag/Delivery/BSGConnectivity.m

```diff
@@ -51,11 +51,13 @@ BOOL BSGConnectivityShouldReportChange(SCNetworkReachabilityFlags flags) {
     #endif
     __block BOOL shouldReport = YES;
     // Check if the reported state is different from the last known state (if any)
-    if ((flags & importantFlags) != (bsg_current_reachability_state & importantFlags)) {
+    SCNetworkReachabilityFlags newFlags = flags & importantFlags;
+    SCNetworkReachabilityFlags oldFlags = bsg_current_reachability_state & importantFlags;
+    if (newFlags != oldFlags) {
         // When first subscribing to be notified of changes, the callback is
         // invoked immmediately even if nothing has changed. So this block
         // ignores the very first check, reporting all others.
-        if (bsg_current_reachability_state == 0) {
+        if (bsg_current_reachability_state == -1) {
```

Comment:
For this to work, don't we need to initialize `bsg_current_reachability_state` to -1 to catch the first call after registering?

Reply:
Yes - this was a local change that didn't make its way into this changeset, I've updated the PR.